pkg/flow/internal/controller/component.go

```diff
@@ -169,6 +170,12 @@ func getExportsType(reg component.Registration) reflect.Type {
 // ID returns the component ID of the managed component from its River block.
 func (cn *ComponentNode) ID() ComponentID { return cn.id }
 
+// Label returns the label for the block or "" if none was specified.
+func (cn *ComponentNode) Label() string { return cn.label }
+
+// NodeType returns the component's type, i.e. `local.file.test` returns `local.file`.
+func (cn *ComponentNode) NodeType() string { return cn.nodeType }
```

Comment:
Since we're aligning on terminology, this should probably be renamed to `ComponentName()` rather than `NodeType()`